src/main/java/com/feedzai/commons/sql/abstraction/engine/DatabaseEngine.java

```diff
@@ -614,15 +623,18 @@ void createPreparedStatement(final String name, final String query, final int ti
 
     /**
      * Creates an iterator for the given SQL expression overriding the configured query
-     * timeout (see {@link PdbProperties#getSelectQueryTimeout()}).
+     * timeout (see {@link PdbProperties#getSelectQueryTimeout()}) if the underlying engine
+     * supports query timeouts.
      *
      * @param query     The expression that represents the query.
      * @param fetchSize The number of rows to fetch each time.
      * @param readTimeoutOverride The query timeout to use.
      * @return An iterator for the results of the given SQL expression.
      * @throws DatabaseEngineException If a database access error occurs.
      */
-    ResultIterator iterator(final Expression query, final int fetchSize, final int readTimeoutOverride) throws DatabaseEngineException;
+    default ResultIterator iterator(final Expression query, final int fetchSize, final int readTimeoutOverride) throws DatabaseEngineException {
+        return iterator(query, fetchSize);
```

Comment:
since `translate` is available as an interface method, you could instead use what the abstract engine is using:
```
        return iterator(translate(query), fetchSize, readTimeoutOverride);
```